presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaSessionProperties.java

```diff
@@ -20,19 +20,24 @@
 import javax.inject.Inject;
 
 import java.util.List;
+import java.util.Set;
 
 public final class KafkaSessionProperties
 {
     private static final String TIMESTAMP_UPPER_BOUND_FORCE_PUSH_DOWN_ENABLED = "timestamp_upper_bound_force_push_down_enabled";
     private final List<PropertyMetadata<?>> sessionProperties;
 
     @Inject
-    public KafkaSessionProperties(KafkaConfig kafkaConfig)
+    public KafkaSessionProperties(KafkaConfig kafkaConfig, @ForKafka Set<List<PropertyMetadata<?>>> extraProperties)
```

Comment:
Can we have something like a `SessionPropertyProvider` ? So we can inject a `Set<SessionPropertyProvider>` and finally flatten when we create the Connector ? `Set<List<PropertyMetadata>>` looks a bit verbose ? 
@kokosing your insights ?